clientconn.go

```diff
@@ -177,17 +174,17 @@ func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *
 		cc.csMgr.channelzID = cc.channelzID
 	}
 
-	if !cc.dopts.insecure {
-		if cc.dopts.copts.TransportCredentials == nil && cc.dopts.copts.CredsBundle == nil {
-			return nil, errNoTransportSecurity
-		}
-		if cc.dopts.copts.TransportCredentials != nil && cc.dopts.copts.CredsBundle != nil {
-			return nil, errTransportCredsAndBundle
-		}
-	} else {
-		if cc.dopts.copts.TransportCredentials != nil || cc.dopts.copts.CredsBundle != nil {
-			return nil, errCredentialsConflict
-		}
+	if cc.dopts.copts.TransportCredentials == nil && cc.dopts.copts.CredsBundle == nil {
+		return nil, errNoTransportSecurity
+	}
+	if cc.dopts.copts.TransportCredentials != nil && cc.dopts.copts.CredsBundle != nil {
+		return nil, errTransportCredsAndBundle
+	}
+	transportCreds := cc.dopts.copts.TransportCredentials
+	if transportCreds == nil {
+		transportCreds = cc.dopts.copts.CredsBundle.TransportCredentials()
+	}
+	if transportCreds != nil && transportCreds.Info().SecurityProtocol == "insecure" {
```

Comment:
`transportCreds != nil` == `true`

Reply:
Why is `transportCreds != nil` not sufficient?

Reply:
Sorry, I was pointing out that it is unnecessary.

I suppose the bundle could have a `nil` `TransportCredentials()` in it. But then we should call that an error, too (on its own), right?

Reply:
Yeah, that was exactly the case for which I added the `nil` check.

Hmm, since we don't allow not configuring a transport credentials, I think it is valid to error out when we have a bundle with `nil` transport creds. But the `Bundle` interface does not document that it should not contain a `nil` transport credentials. 

Do you think this will be a behavior change that can break people or you think it is better to enforce it?

Reply:
Nobody should be taking advantage of that loophole.  Bundles are experimental, as well.  Let's update the documentation to say transport credentials are required (and may be insecure.NewCredentials() if needed), and enforce it here as well.

Reply:
Done. Added the constraint, related documentation and test.